plugins/registry.js

```diff
@@ -438,6 +440,21 @@ export default class PluginRegistry {
         registerPluginTranslationsSource(this.id, getTranslationsForLocale);
     }
 
+    // Register a admin console definitions override function.
+    // - func - A function a function that recieve the admin console config
+    // definitions and return a new version of it, which is used for build the
+    // admin console.
+    // Each plugin can only register a function, if you reguster multiple functions the last one will be used.
+    registerAdminConsolePlugin(func) {
```

Comment:
to help plugin-writers, it would help tremendously if we gave an example of what's passed in the function and a real world extension example of the function and what to return from it. Would be worth mentioning the function can mutate the passed in arg admin definition as it's a copy of the original

Reply:
+1 on documenting what is the expected function signature and what is the expected behavior as well. This is the documentation that is published in our [plugin reference site](https://developers.mattermost.com/extend/plugins/webapp/reference/#registry).

Reply:
oh neat. didn't know we had that documentation. Sad to see the `*` indents are off :(
```
/**
    * Register a component fixed to the bottom of the post message.
    * Accepts a React component. Returns a unique identifier.
*/
```

Reply:
A realistic example of "input" would be just a lot of data. Basically what that function normally receive is the entire declarative admin definition and because it is a "low-level" function for plugins I would expect you to know what is the admin definitions structure (if not, you should be using the other functions that we going to develop on top of this).